sdk/go/common/util/env/env.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A small library for creating consistent and documented environmental variable accesses.
+//
+// Public environmental variables should be declared as a module level variable.
+package env
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+)
+
+var Experimental = Bool("EXPERIMENTAL", "Enable experimental options and commands")
+
+type Var struct {
+	name        string
+	Value       Value
+	Description string
+	options     options
+}
+
+const Prefix = "PULUMI"
```

Comment:
Since this appears in the public API, it might be simpler to write this as `PULUMI_` and not have to join it elsewhere. Just a little more explicit for readers of the API.